### PR TITLE
SOT-128: RC hash

### PR DESCRIPTION
**Pull request by bjergsen243**
# Check List

- [x] Did you check your grammar / reader proof your document using ChatGPT?
- [x] Don't forget to squash commits into meaningful chunks before merging
- [x] After merging this, please update your task status!
- [x] Did you verify that your math formulas are rendered correctly?

### CODE REVIEW

File: docs/reinforced-concrete-hash.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+Other helpful resources:
+
++ [Reinforced Concrete Hash Slide](
+https://www.slideshare.net/AlexPruden/zkstudyclub-plonkup-reinforced-concrete-luke-pearson-joshua-fitzgerald-dmitry-khovratovich) .
++ [Reinforced Concrete and Plonkup Video](https://www.youtube.com/watch?v=Pnc9J7uQgqs&ab_channel=ZeroKnowledge).
++ [Reinforced Concrete by Dmitry Khovratovich](https://www.youtube.com/watch?v=SXnb7T9YATs&ab_channel=ZeroKnowledge).
+
+## Overview
+
+The Reinforced Concrete (RC) hash function works in the sponge framework, transforming a fixed-
```
Comment (zk-steve) [inline docs/reinforced-concrete-hash.md:19]:
```suggestion
The Reinforced Concrete (RC) hash function works in the sponge framework, transforming a fixed-length
```

File: docs/reinforced-concrete-hash.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+length bijective function into a variable-length hash.
+
+The RC permutation, shown in figure below, is a modified 7-round [SP network](
```
Comment (zk-steve) [inline docs/reinforced-concrete-hash.md:22]:
```suggestion
**The RC permutation**, shown in figure below, is a modified 7-round [SP network](
```

File: docs/reinforced-concrete-hash.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
+  
+# Reinforced Concrete
+
+A fast and efficient [zk-friendly hash](../terms/zk-friendly-hash.md) function that leverages lookup gates to ensure security.
```
Comment (VanhGer) [inline docs/reinforced-concrete-hash.md:4]: file not found.

File: docs/reinforced-concrete-hash.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
++ [Reinforced Concrete and Plonkup Video](https://www.youtube.com/watch?v=Pnc9J7uQgqs&ab_channel=ZeroKnowledge).
++ [Reinforced Concrete by Dmitry Khovratovich](https://www.youtube.com/watch?v=SXnb7T9YATs&ab_channel=ZeroKnowledge).
+
+## Overview
+
+The Reinforced Concrete (RC) hash function works in the sponge framework, transforming a fixed-
+length bijective function into a variable-length hash.
+
+The RC permutation, shown in figure below, is a modified 7-round [SP network](
+https://en.wikipedia.org/wiki/Substitution%E2%80%93permutation_network), with elements from $\mathbb{F}_ p^3$ for a prime number $p$.
```
Comment (VanhGer) [inline docs/reinforced-concrete-hash.md:23]: Please explain what is SP network instead of giving only a reference link to Wikipedia

**Review by zk-steve — APPROVED** [review]

**Pull request merged**